### ISSUE TITLE
fromJS meet class instance will stop?

### ISSUE THREAD

**Issue by xp44mm** [facebook/immutable-js#1555]
```javascript
    test('test fromJS', () => {
        const obj = fromJS({ a: '123', b: '234', air: new AirViewModel() })
        console.log(obj)
    })
```
the output is:
```javascript
 Map { "a": "123", "b": "234", "air": [object Object] }
```

but the `air` should be as plain object? 
```javascript
 Map { "a": "123", "b": "234", "air": Map{...} }
```

**Comment by bhowell2** [comment]
Why should `AirViewModel` be made into a map? How is `fromJS` supposed to know what to do with it? Documentation specifically says it converts plain JS objects and arrays.

**Comment by xp44mm** [comment]
yes, thank you! the `fromJS` maybe should rename to `fromPlainJS`.

**Comment by conartist6** [comment]
I think actually there could be an improvement that would allow you to handle this case nicely.

Currently `toJS` only touches things that are "data structures", which does not include class instances.

What if, upon encountering a class instance, `toJS` called the instance's `toJSON()` method if it were present?

Then @xp44mm could call `fromJS(Seq(...).toJS())`. This looks a bit weird but I think conceptually it makes sense given the mix of serializable and unserializable data.
